third_party/move/move-model/src/ty.rs

```diff
@@ -1403,22 +1474,12 @@ impl TypeUnificationError {
                 format!("mutability mismatch ({} != {})", pr(*k1), pr(*k2))
             },
             TypeUnificationError::ConstraintUnsatisfied(_, ty, order, constr) => match constr {
-                Constraint::SomeNumber(options) => {
-                    let all_ints = PrimitiveType::all_int_types()
-                        .into_iter()
-                        .collect::<BTreeSet<_>>();
-                    let options_str = if options == &all_ints {
-                        "integer".to_owned()
-                    } else {
-                        options
-                            .iter()
-                            .map(|p| Type::new_prim(*p).display(display_context).to_string())
-                            .join("|")
-                    };
+                Constraint::SomeNumber(_) => {
+                    let options_str = constr.display(display_context);
                     let type_str = ty.display(display_context).to_string();
                     let (expected, actual) = match order {
-                        WideningOrder::Join | WideningOrder::LeftToRight => (type_str, options_str),
-                        WideningOrder::RightToLeft => (options_str, type_str),
+                        WideningOrder::Join | WideningOrder::LeftToRight => (options_str, type_str),
```

Comment:
Related to @brmataptos's comment on the reversed error message , why is the order reversed here?